### ISSUE TITLE
Provide a visual indication for entries that contain unsaved changes

### ISSUE THREAD

**Issue by magkopian** [keepassxreboot/keepassxc#1162]
It would be very useful if there was a way to easily tell which entries contains unsaved changes.

## Expected Behavior
The program should have a way of informing the user about which entries contain unsaved changes. This way the user has the opportunity of inspecting the particular entries, and checking for any accidental changes before saving the database.

## Current Behavior
There is currently no way of telling which entries have been changed since the last time the database was saved.

## Possible Solution
Change the font style of the column fields of the entry to italic, or the font color to something else (e.g. red), or append an asterisk to the title field of the entry, in order to indicate that the entry contains unsaved changes, until the database is saved. And possibly, also do the same for the names for groups that contains such entries, in order to make their discovery easier. A toolbar button for listing all the entries that contain unsaved changes, could also be a possible solution to the discovery of such entries.

## Context
In the past there were several occasions, where I noticed the familiar asterisk on the title bar indicating that something has been modified, but I had absolutely no idea what that was. The result was that I was unsure whether I should save the changes or not.

## Debug Info
Libraries:
KeePassXC - Version 2.2.2
Revision: 6d46717

Libraries:
- Qt 5.7.1
- libgcrypt 1.7.6-beta

Operating system: Debian GNU/Linux 9 (stretch)
CPU architecture: x86_64
Kernel: linux 4.9.0-4-amd64

Enabled extensions:
- KeePassHTTP
- Auto-Type
- YubiKey

**Comment by harwoeck** [comment]
This feature would be nice to have 👏  But I don't agree with just highlighting changes. If you have a lot of folders and have to browse every folder before saving, it can get really annoying. I would suggest:

1. Highlight as mentioned by @magkopian 
2. When "Save database" is called, open a pop-up window listing all changes. Simple put, it's like a `git status` where i can see additions, deletions, renaming, changes, and so on.

Also think about providing a checkbox "Don't show changes when saving" to all users who are not interested in this feature.

Let me know what you think 👍

**Comment by TheZ3ro** [comment]
We have autosave enabled by default now, so this seems a little useless IMHO

**Comment by droidmonkey** [comment]
Nice idea

**Comment by gameguy95** [comment]
@TheZ3ro It seems to me that autosave is the exact opposite of what this user is requesting. He wants the ability to review changes in order to determine if he changed something by accident, which would give him the ability to revert the change if required. If the user has autosave enabled, even accidental changes would be saved automatically without warning and would only be reversible if the user has a recent backup of their keepass database.

**Comment by harwoeck** [comment]
@gameguy95 Absolutely correct! 👍

**Comment by TheZ3ro** [comment]
@gameguy95 Exactly, since autosave  it's the default now, every change made is automatically saved. A known change is reversible through the entry history.

Anyway, I think it's a good candidate for the wishlist

**Comment by ShapeShifter499** [comment]
KeePassX has a backup feature that saves a copy of recently changed entries to a 'backup' group. Can we get that here?

**Comment by droidmonkey** [comment]
There is no use for this, we have a history for each entry that is customizable in the database settings. You can store as many versions of an entry as you want.

**Comment by ShapeShifter499** [comment]
@droidmonkey Oh sorry, new user. Please disregard my previous comment.

**Comment by ShapeShifter499** [comment]
@droidmonkey That history is stored in the database file such that it transfers to new computers right?

**Comment by fredericschulz** [comment]
So is this issue now to be implemented only for when autosaving is deactivated or whats the idea?

**Comment by droidmonkey** [comment]
Correct, this type of indicator would not be necessary or desired when autosave is enabled.
